### PR TITLE
Feature/fix filtering redacted

### PR DESCRIPTION
**Pull request by ganfra**
There was an issue with the filtering condition of redacted events in timeline. It was causing this issue => https://github.com/vector-im/element-android/issues/2094

Also, filtering was not working for changes on already built events.

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt [hunk 1]
```diff
@@ -333,11 +333,19 @@ internal class DefaultTimeline(
...
             builtEvents[builtIndex]?.let { te ->
-                builtEvents[builtIndex] = builder(te)
+                val rebuiltEvent = builder(te)
+                // If rebuilt event is filtered its returned as null and should be removed.
+                if (rebuiltEvent == null) {
+                    builtEventsIdMap.remove(eventId)
+                    builtEventsIdMap.entries.filter { it.value > builtIndex }.forEach { it.setValue(it.value - 1) }
+                    builtEvents.removeAt(builtIndex)
+                } else {
+                    builtEvents[builtIndex] = builder(te)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt:347]: Use `rebuiltEvent`?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt [hunk 2]
```diff
@@ -775,8 +784,7 @@ internal class DefaultTimeline(
             }
             if (!filterEdits) return@filter false
 
-            val filterRedacted = !settings.filters.filterRedacted || it.root.isRedacted()
-
+            val filterRedacted = !settings.filters.filterRedacted || it.root.isRedacted().not()
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt:787]: For clarity and coherency (I had to read it twice) we should write:

```
val filterRedacted = settings.filters.filterRedacted && it.root.isRedacted()

return !filterRedacted
```

Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt:787]: Yup totally, my brain was off :O

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt (hunk 1 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt:347]: Yes good catch

**Pull request merged**